be/src/util/owned_slice.h

```diff
@@ -0,0 +1,60 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include "gutil/macros.h"
+
+namespace doris {
+
+struct Slice;
+
+class OwnedSlice {
+
+public:
+    Slice slice = nullptr;
+
+    OwnedSlice(){}
+
+    OwnedSlice(Slice _slice): slice(_slice){}
+
+    ~OwnedSlice(){
+        if (slice != nullptr) {
+            delete[] slice.data;
+        }
+    }
+
+    OwnedSlice(OwnedSlice&& src): slice(src.slice) {
+        src.slice = nullptr;
+    }
+
+    OwnedSlice& operator =(OwnedSlice&& src) {
+        if (this != &src) {
```

Comment:
why not swap this and src?

Reply:
There is assignment operation in std::swap,so call swap in OwnedSlice's move assignment operator would trigger OwnedSlice's move assignment operator again,and then fall into infinite loop